packages/main/src/plugin/authentication.ts

```diff
@@ -75,6 +78,64 @@ export interface SessionRequestInfo {
 }
 
 export class AuthenticationImpl {
+  async showAccountsMenu(x: number, y: number): Promise<void> {
+    const template: (MenuItemConstructorOptions | MenuItem)[] = [
+      {
+        label: 'Manage authentication',
+        click: (): void => {
+          this.apiSender.send('navigate', {
+            page: 'authentication-providers',
+          });
+        },
+      },
+      {
+        type: 'separator',
+      } as MenuItem,
+    ];
+
+    this.getSessionRequests().forEach(request => {
+      const provider = this._authenticationProviders.get(request.providerId);
+      if (!provider) return;
+      const providerLabel = provider.label ? provider.label : provider.id;
+      template.push({
+        label: `Sign in with ${providerLabel} to use ${request.extensionLabel}`,
+        click: () => {
+          void this.executeSessionRequest(request.id);
+        },
+      });
+    });
+
+    const providers = Array.from(this._authenticationProviders.values());
+
+    const menuItemPromises = providers.map(meta =>
+      meta.provider.getSessions().then(sessions => {
+        sessions.forEach(session => {
+          template.push({
+            label: `${session.account.label} (${meta.label})`,
+            submenu: [
+              {
+                label: 'Sign out',
+                click: (): void => {
+                  void this.signOut(meta.id, session.id);
```

Comment:
should not have the void prefix there

Reply:
I think there should not have any direct calls using this as it's making harder to move the code later to the renderer part.
It should delegate calls to the IPC

Reply:
void is for eslint to stop complaining. it ether this or ```this.signOut(meta.is, session.id).catch((err) => console.log(err))```

Reply:
yes please use the catch, we should not bypass/fake the linter.

Reply:
Updated